web/ui/src/Services/ServiceDetailsController.js

```diff
@@ -297,10 +298,13 @@
                 var host = publicEndpoint.Name.indexOf('.') === -1 ? publicEndpoint.Name + "." + $scope.defaultHostAlias : publicEndpoint.Name;
                 return location.protocol + "//" + host + port;
             } else if(publicEndpoint.type === "port"){
-                // TODO - get IP
-                var host = $scope.defaultHostAlias;
-                // Port public endpoint port listeners are always on http
-                return "http://" + host + ":" + publicEndpoint.PortNumber;
+                if(publicEndpoint.PortNumber.startsWith(":")){
+                    var host = $scope.defaultHostAlias;
+                    // Port public endpoint port listeners are always on http
+                    return "http://" + host + publicEndpoint.PortNumber;
+                }else{
+                    return "http://" + publicEndpoint.PortNumber;
+                }
```

Comment:
This code is looking for `PortNumber` property, but I think that property was renamed to `PortAddr`?